websites/P/picoCTF/presence.ts

```diff
@@ -0,0 +1,181 @@
+const presence = new Presence({
+		clientId: "989558631823474748",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const privacy = await presence.getSetting<boolean>("privacy"),
+		presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp,
+		};
+
+	if (document.location.hostname === "picoctf.org") {
+		if (document.location.pathname.includes("/learning_guides/")) {
+			presenceData.details = "Reading a guide:";
+			presenceData.state = `${document.location.pathname
+				.split("/")[2]
+				.replaceAll("-", " ")
+				.slice(6, -4)}`;
+		} else if (document.location.pathname.includes("/pdfs/")) {
+			presenceData.details = document.location.pathname.includes(
+				"Sponsorship_Flyer"
+			)
+				? "Reading about sponsoring picoCTF"
+				: "Reading a research paper";
+		} else if (document.location.pathname === "/get_started.html") {
+			const viewing =
+				document.querySelectorAll(
+					'ul#competition-tabs li a:not([aria-selected="false"])'
+				)[0].innerHTML === "User Guide"
+					? "Viewing the user guide"
+					: "Viewing the FAQ";
+
+			presenceData.details = viewing;
+
+			if (!privacy && viewing === "Viewing the user guide") {
+				presenceData.state = Array.from(
+					document.querySelectorAll(
+						'div.active button:not([aria-selected="false"]) span'
+					),
+					el => el.innerHTML
+				)
+					.splice(
+						document
+							.querySelectorAll(
+								'ul#guide-tabs li a:not([aria-selected="false"])'
+							)[0]
+							.innerHTML.includes("Learner")
+							? 0
+							: 2,
+						2
+					)
+					.join(": ");
+			}
+		} else {
+			switch (document.location.pathname) {
+				case "/":
+					presenceData.details = "Viewing home page";
+					break;
+				case "/resources.html":
+					presenceData.details = "Viewing learning resources";
+					break;
+				case "/community.html":
+					presenceData.details = "Viewing community page";
+					break;
+				case "/competitions/past.html":
+					presenceData.details = "Viewing past competitions";
+					break;
+				case "/privacy.html":
+					presenceData.details = "Reading privacy statement";
+					break;
+				case "/terms.html":
+					presenceData.details = "Reading terms of service";
+					break;
+				case "/about.html":
+					presenceData.details = "Reading about picoCTF";
```

Comment:
Update this for other references of the service name as it would be repeating the service name 
```suggestion
					presenceData.details = "Reading the about page";
```

Reply:
Resolved in 6c865a0